op-nat/config.go

```diff
@@ -1,55 +1,65 @@
 package nat
 
 import (
-	"context"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"math/big"
-	"math/rand"
 	"os"
+	"path/filepath"
 	"strconv"
 
 	"github.com/urfave/cli/v2"
 
 	"github.com/ethereum-optimism/infra/op-nat/flags"
 	"github.com/ethereum-optimism/infra/op-nat/network"
-	"github.com/ethereum-optimism/infra/op-nat/wallet"
 	"github.com/ethereum/go-ethereum/log"
 )
 
 type Config struct {
 	// Network config
-	SC         SuperchainManifest
-	Validators []Validator
-
-	Wallets []*wallet.Wallet
-
-	// Networks
+	SC  SuperchainManifest
 	L1  *network.Network
 	L2A *network.Network
 
+	// Test config
+	TestDir         string
+	ValidatorConfig string
+	TargetGate      string
+
 	Log log.Logger
 }
 
-func NewConfig(ctx *cli.Context, log log.Logger, validators []Validator) (*Config, error) {
+// NewConfig creates a new Config instance
+func NewConfig(ctx *cli.Context, log log.Logger, testDir string, validatorConfig string, gate string) (*Config, error) {
 	// Parse flags
 	if err := flags.CheckRequired(ctx); err != nil {
 		return nil, fmt.Errorf("missing required flags: %w", err)
 	}
+	if testDir == "" {
+		return nil, errors.New("test directory is required")
+	}
+	if validatorConfig == "" {
+		return nil, errors.New("validator config path is required")
+	}
+	if gate == "" {
+		return nil, errors.New("gate is required")
+	}
 
-	// Parse kurtosis-devnet manifest
-	manifest, err := parseManifest(ctx.String(flags.KurtosisDevnetManifest.Name))
+	// Get absolute paths
+	absTestDir, err := filepath.Abs(testDir)
 	if err != nil {
-		return nil, fmt.Errorf("failed to parse kurtosis-devnet manifest: %w", err)
+		return nil, fmt.Errorf("failed to get absolute path for test dir: %w", err)
+	}
+	absValidatorConfig, err := filepath.Abs(validatorConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get absolute path for validator config: %w", err)
 	}
 
-	wallets := []*wallet.Wallet{}
-	for i, w := range manifest.L1.Wallets {
-		w, err := wallet.NewWallet(w.PrivateKey, fmt.Sprintf("user-key-%s", i))
-		if err != nil {
-			log.Warn("error creating wallet: %w", err)
-		}
-		wallets = append(wallets, w)
+	// Parse kurtosis-devnet manifest
```

Comment:
nit: it's meant to *not* be kurtosis-specific :)

Reply:
Definitely. I've since removed this actually. The assumption now is that the tests _must_ use `devnet-sdk` and it will handle the manifest-parsing.